src/structures/idt.rs

```diff
@@ -779,34 +779,47 @@ impl<F> Entry<F> {
     }
 }
 
-macro_rules! impl_set_handler_fn {
-    ($h:ty) => {
+#[cfg(all(feature = "instructions", feature = "abi_x86_interrupt"))]
+impl<F: HandlerFuncType> Entry<F> {
+    /// Set the handler function for the IDT entry and sets the present bit.
+    ///
+    /// For the code selector field, this function uses the code segment selector currently
+    /// active in the CPU.
+    ///
+    /// The function returns a mutable reference to the entry's options that allows
+    /// further customization.
+    ///
+    /// This method is only usable with the `abi_x86_interrupt` feature enabled. Without it, the
+    /// unsafe [`Entry::set_handler_addr`] method has to be used instead.
+    #[inline]
+    pub fn set_handler_fn(&mut self, handler: &F) -> &mut EntryOptions {
```

Comment:
For backwards compatibility, `set_handler_fn` needs to take the handler by value, not by reference.

Reply:
Gotcha. I didn't realize that function pointers are `Copy`, so it doesn't matter if it's owned or not

Reply:
Just for my own understanding (I'm fairly new to Rust), why does `set_handler_fn` take the handler function by value? If there's nothing in `set_handler_fn` that requires the function to be moved, isn't it better to take a reference so that the caller can keep using the argument? Or is it specifically because it's a function, there's never a reason to pass a function by reference, and you should always Copy as needed?

Reply:
Ya this can be a confusing thing initially, especially because the function Traits (`Fn`, `FnMut`, `FnOnce`) are often handled differently.

A type like `fn(u8) -> u8` or `extern "x86-interrupt" fn(InterruptStackFrame) -> !` is a function _pointer_, so it references a specific function. It this way, it is similar to `&T` or `*const T` in that it is:
 - pointer sized
 - a shared reference to something immutable.

So for the same reason we usually don't use `&&T` or `&*const T`, using `&fn(u8) -> u8` is uncommon.